### DISCUSSION TITLE
AttributeError: module 'syft' has no attribute 'join_duet'

### DISCUSSION THREAD

**Discussion by gonzalo-munillag** [OpenMined/courses#298] [Course: Foundations of Private Computation]
Hi!

After I successfully installed the requirements and used the pip install git command to install the newest version of Pysyft, and after I restarted the kernels of the notebooks, and after I restarted the computer (All this with unsetting in the corresponding terminal environment the proxies I have in my company computer, so that the notebooks should not have a proxy issue) I get the error below.
I found the same error at the end of this [thread](https://github.com/OpenMined/courses/discussions/288)

But I do not know unfortunately what else to try. Do you guys know the solution to the problem?

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-2-3bdb486575e7> in <module>
----> 1 duet = sy.join_duet('4814df40925c14ae8f950342e3e8d776', network_url="http://ec2-18-216-8-163.us-east-2.compute.amazonaws.com:5000/")

AttributeError: module 'syft' has no attribute 'join_duet'

**Comment by deepak121993** [comment]
Hi Gonzalo,
There is versioning issue. By default the Syft installable not supporting this feature. Check out the below conversation.
[https://github.com//discussions/273#discussioncomment-494352]

You need to follow below steps:
1. pip uninstall syft
2. pip install git+https://github.com/OpenMined/PySyft@dev#egg=syft

or what I did, I had hardcoded the version itself as mentioned in comment:

pip install syft-0.3.0.post0.dev754+g65a4aa120
This works for me.


Thanks,
Deepak

**Reply by gonzalo-munillag** [reply]
Hey! Thanks for taking the time, I did that but now my system went bunkers
<img width="1077" alt="Screenshot 2021-03-24 at 13 07 26" src="https://user-images.githubusercontent.com/57599753/112308033-f13f1a80-8ca1-11eb-8802-49abad61d503.png">

**Reply by gonzalo-munillag** [reply]
And yes, I restarted everything, and also before that I updated Jupiter lab and all

**Reply by gonzalo-munillag** [reply]
And the syft requirement about having nest_asyncio is already satisfied
<img width="726" alt="Screenshot 2021-03-24 at 13 17 18" src="https://user-images.githubusercontent.com/57599753/112309183-58110380-8ca3-11eb-917d-58f085fbc7f3.png">

**Comment by gonzalo-munillag** [comment]
In the end, I created a new environment, now it works. Nonetheless, what happened in the other environment is still a mystery. Probably something to do with syft versions.
